### ISSUE TITLE
dtype of `np.random.choice` differs with presence of `p` on windows

### ISSUE THREAD

**Issue by jcrist** [numpy/numpy#9867]
The dtype of the output of `np.random.choice` with a scalar `a` is different if `p` is provided on windows:

```python
>>> import numpy as np
>>> p = np.array([0.5, 0.5])
>>> np.random.choice(2, size=1).dtype
dtype('int32')
>>> np.random.choice(2, p=p, size=1).dtype
dtype('int64')
>>> np.__version__
1.13.1
```

I'd expect the output dtype to be consistent whether `p` is present or absent. Since the `dtype` of `arange` on win32 is `int32`, the `int64` dtype output is probably the incorrect one.

**Comment by bashtage** [comment]
This difference occurs at

https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/mtrand.pyx#L1155

since `searchsorted` returns `int64` irrespective of platform.

**Comment by bashtage** [comment]
The return type for `searchsorted` is `NPY_INTP`  here:

https://github.com/numpy/numpy/blob/c2372aff1ea85e8e1d64d8b5ced7e362fd0f4a5a/numpy/core/src/multiarray/item_selection.c#L1732

**Comment by bashtage** [comment]
@mattip  This is fixed in Generator but not in RandomState.  Should it be fixed in RandomState or left as it was in 1.16?

**Comment by mattip** [comment]
I guess this is a bug, and should be fixed, unless it breaks something else.
